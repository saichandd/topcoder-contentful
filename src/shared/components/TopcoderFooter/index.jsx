import moment from 'moment';
import PT from 'prop-types';
import React from 'react';

import { config } from 'topcoder-react-utils';

import FacebookIcon from './icons/facebook.svg';
import GooglePlusIcon from './icons/google_plus.svg';
import LinkedInIcon from './icons/linkedin.svg';
import TwitterIcon from './icons/twitter.svg';

import './style.scss';

function Link({
  children,
  to,
}) {
  return (
    <li styleName="link">
      <a href={to}>
        {children}
      </a>
    </li>
  );
}

Link.propTypes = {
  to: PT.string.isRequired,
  children: PT.node.isRequired,
};

export default function TopcoderFooter() {
  const base = config.URL.BASE;
  const currentYear = moment().year();
  return (
    <div styleName="footer">
      <nav>
        <ul>
          <Link to={`${base}/sitemap`}>
SITE MAP
          </Link>
          <Link to={`${base}/about`}>
ABOUT US
          </Link>
          <Link to={`${config.URL.HELP}/hc/en-us/articles/219069687-Contact-Support`}>
CONTACT US
          </Link>
          <Link to={config.URL.HELP}>
HELP CENTER
          </Link>
          <Link to={`${base}/community/how-it-works/privacy-policy/`}>
PRIVACY POLICY
          </Link>
          <Link to={`${base}/community/how-it-works/terms/`}>
TERMS
          </Link>
        </ul>
      </nav>
      <div styleName="social-links">
        <hr />
        <a href="https://www.facebook.com/topcoder" target="fbwindow">
          <FacebookIcon />
        </a>
        <a href="http://www.twitter.com/topcoder" target="twwindow">
          <TwitterIcon />
        </a>
        <a href="https://www.linkedin.com/company/topcoder" target="liwindow">
          <LinkedInIcon />
        </a>
        <a href="https://plus.google.com/u/0/b/104268008777050019973/104268008777050019973/posts" target="gpwindow">
          <GooglePlusIcon />
        </a>
      </div>
      <p styleName="copyright-notice">
        {`Â© ${currentYear} Topcoder. All Rights Reserved`}
      </p>
    </div>
  );
}
