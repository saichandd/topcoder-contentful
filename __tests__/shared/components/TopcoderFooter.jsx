import React from 'react';
import Rnd from 'react-test-renderer';
import TopcoderFooter from 'components/TopcoderFooter';

test('Matches shallow shapshot', () => {
  const rnd = Rnd.create(<TopcoderFooter />);
  expect(rnd.toJSON()).toMatchSnapshot();
});
