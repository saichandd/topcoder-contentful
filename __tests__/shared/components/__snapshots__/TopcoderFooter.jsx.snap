// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-TopcoderFooter-___style__footer___28yQ6"
>
  <nav>
    <ul>
      <li
        className="src-shared-components-TopcoderFooter-___style__link___3-nzm"
      >
        <a
          href="https://www.topcoder-dev.com/sitemap"
        >
          SITE MAP
        </a>
      </li>
      <li
        className="src-shared-components-TopcoderFooter-___style__link___3-nzm"
      >
        <a
          href="https://www.topcoder-dev.com/about"
        >
          ABOUT US
        </a>
      </li>
      <li
        className="src-shared-components-TopcoderFooter-___style__link___3-nzm"
      >
        <a
          href="https://help.topcoder-dev.com/hc/en-us/articles/219069687-Contact-Support"
        >
          CONTACT US
        </a>
      </li>
      <li
        className="src-shared-components-TopcoderFooter-___style__link___3-nzm"
      >
        <a
          href="https://help.topcoder-dev.com"
        >
          HELP CENTER
        </a>
      </li>
      <li
        className="src-shared-components-TopcoderFooter-___style__link___3-nzm"
      >
        <a
          href="https://www.topcoder-dev.com/community/how-it-works/privacy-policy/"
        >
          PRIVACY POLICY
        </a>
      </li>
      <li
        className="src-shared-components-TopcoderFooter-___style__link___3-nzm"
      >
        <a
          href="https://www.topcoder-dev.com/community/how-it-works/terms/"
        >
          TERMS
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="src-shared-components-TopcoderFooter-___style__social-links___2G4Ia"
  >
    <hr />
    <a
      href="https://www.facebook.com/topcoder"
      target="fbwindow"
    >
      <svg
        height="28"
        viewBox="0 0 27 28"
        width="27"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon facebook
        </title>
        <path
          d="M24 .5H3c-1.65 0-3 1.35-3 3v21c0 1.651 1.35 3 3 3h10.5V17h-3v-3.713h3v-3.074c0-3.247 1.818-5.527 5.649-5.527l2.705.003v3.908h-1.796C18.567 8.597 18 9.716 18 10.754v2.535h3.852L21 17h-3v10.5h6c1.65 0 3-1.349 3-3v-21c0-1.65-1.35-3-3-3z"
          fill="#A3A3AD"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <a
      href="http://www.twitter.com/topcoder"
      target="twwindow"
    >
      <svg
        height="23"
        viewBox="0 0 28 23"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon twitter
        </title>
        <path
          d="M24.774 6.083c.012.243.016.489.016.732 0 7.485-5.695 16.113-16.11 16.113-3.199 0-6.174-.938-8.68-2.546.444.053.894.08 1.35.08 2.655 0 5.096-.906 7.032-2.422a5.668 5.668 0 0 1-5.289-3.932 5.787 5.787 0 0 0 2.557-.097A5.668 5.668 0 0 1 1.11 8.456v-.07c.763.424 1.637.679 2.564.709a5.662 5.662 0 0 1-1.751-7.561A16.073 16.073 0 0 0 13.593 7.45a5.72 5.72 0 0 1-.146-1.291A5.66 5.66 0 0 1 19.11.5c1.628 0 3.1.686 4.133 1.786 1.289-.255 2.501-.725 3.596-1.374a5.683 5.683 0 0 1-2.49 3.132 11.326 11.326 0 0 0 3.252-.89 11.435 11.435 0 0 1-2.826 2.929z"
          fill="#A3A3AD"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <a
      href="https://www.linkedin.com/company/topcoder"
      target="liwindow"
    >
      <svg
        height="28"
        viewBox="0 0 27 28"
        width="27"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon linkedln
        </title>
        <path
          d="M6 3.5c0 1.65-1.05 3-3 3-1.8 0-3-1.35-3-2.85C0 2 1.2.5 3 .5s3 1.35 3 3zm-6 24h6V8H0v19.5zM20.4 8.3c-3.15 0-4.95 1.8-5.7 3h-.15l-.3-2.55h-5.4c0 1.65.15 3.6.15 5.85v12.9h6V16.85c0-.6 0-1.05.15-1.5.45-1.05 1.2-2.4 2.85-2.4 2.1 0 3 1.8 3 4.2V27.5h6V16.4c0-5.55-2.85-8.1-6.6-8.1z"
          fill="#A3A3AD"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <a
      href="https://plus.google.com/u/0/b/104268008777050019973/104268008777050019973/posts"
      target="gpwindow"
    >
      <svg
        height="27"
        viewBox="0 0 28 27"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon google plus
        </title>
        <path
          d="M1.45 7.083c0 2.242.762 3.858 2.262 4.808 1.23.78 2.656.897 3.398.897.18 0 .325-.009.425-.015 0 0-.235 1.506.9 2.994h-.05c-1.968 0-8.385.404-8.385 5.59C0 26.631 5.893 26.9 7.076 26.9c.093 0 .148-.003.148-.003.012 0 .096.003.241.003.759 0 2.72-.093 4.541-.965 2.363-1.125 3.56-3.088 3.56-5.827 0-2.646-1.825-4.221-3.157-5.373-.814-.704-1.517-1.31-1.517-1.899 0-.6.514-1.051 1.163-1.623 1.051-.922 2.044-2.238 2.044-4.725 0-2.186-.289-3.654-2.067-4.585.185-.094.84-.161 1.165-.206.963-.129 2.371-.276 2.371-1.049V.5h-7.02C8.479.503 1.45.758 1.45 7.083zm11.331 13.52c.135 2.109-1.701 3.665-4.46 3.861-2.8.203-5.105-1.036-5.24-3.144-.065-1.014.388-2.004 1.275-2.795.9-.799 2.134-1.294 3.477-1.392.159-.008.316-.018.473-.018 2.593.002 4.349 1.499 4.475 3.488zM10.948 5.639c.689 2.382-.35 4.869-2.008 5.33-.191.052-.387.078-.586.078-1.518 0-3.021-1.51-3.58-3.59-.311-1.164-.285-2.187.072-3.168.35-.967.981-1.617 1.775-1.838.19-.052.388-.079.588-.079 1.831 0 3.01.747 3.74 3.267zM22.835 10.7V6.2h-3.052v4.5h-4.58v3h4.58v4.5h3.052v-4.5h4.58v-3h-4.58z"
          fill="#A3A3AD"
          fillRule="evenodd"
        />
      </svg>
    </a>
  </div>
  <p
    className="src-shared-components-TopcoderFooter-___style__copyright-notice___24f5v"
  >
    Â© 2018 Topcoder. All Rights Reserved
  </p>
</div>
`;
